docs/operators/regexparser.md

```diff
@@ -0,0 +1,286 @@
+
+```xml
+    <property>
+        <name>dt.application.{ApplicationName}.operator.{OperatorName}.port.out.attr.TUPLE_CLASS</name>
+        <value>com.datatorrent.tutorial.regexparser.ServerLog</value>
+    </property>
+```
+
+Below is the example for setting `TUPLE_CLASS` attribute on output port from `Application.java` file of the application.
+
+```java
+RegexParser regexParser = dag.addOperator("regexParser", RegexParser.class);
+dag.setOutputPortAttribute(regexParser.out, Context.PortContext.TUPLE_CLASS, ServerLog.class);
+```
+
+where the value (ServerLog) set above is the expected output POJO class from the operator and example is as below.
+
+```java
+    public class ServerLog
+    {
+      private Date date;
+      private int id;
+      private String signInId;
+      private String ipAddress;
+      private double serviceId;
+      private long accountId;
+      private boolean platform;
+    
+    
+      public int getId()
+      {
+        return id;
+      }
+    
+      public void setId(int id)
+      {
+        this.id = id;
+      }
+    
+      public Date getDate()
+      {
+        return date;
+      }
+    
+      public void setDate(Date date)
+      {
+        this.date = date;
+      }
+    
+      public String getSignInId()
+      {
+        return signInId;
+      }
+    
+      public void setSignInId(String signInId)
+      {
+        this.signInId = signInId;
+      }
+    
+      public String getIpAddress()
+      {
+        return ipAddress;
+      }
+    
+      public void setIpAddress(String ipAddress)
+      {
+        this.ipAddress = ipAddress;
+      }
+    
+      public double getServiceId()
+      {
+        return serviceId;
+      }
+    
+      public void setServiceId(double serviceId)
+      {
+        this.serviceId = serviceId;
+      }
+    
+      public long getAccountId()
+      {
+        return accountId;
+      }
+    
+      public void setAccountId(long accountId)
+      {
+        this.accountId = accountId;
+      }
+    
+      public boolean getPlatform()
+      {
+        return platform;
+      }
+    
+      public void setPlatform(boolean platform)
+      {
+        this.platform = platform;
+      }
+    }
+```
+
+Let us look at how the data gets populated into the POJO using the example `schema`, `splitRegexPattern` and `TUPLE_CLASS` definitions given above.
+
+Consider sample event log as below that matches with the `splitRegexPattern`.
+
+```
+2015-10-01T03:14:49.000-07:00 lvn-d1-dev DevServer[9876]: INFO: [EVENT][SEQ=248717] 2015:10:01:03:14:49 101 sign-in_id=11111@psop.com ip_address=1.1.1.1  service_id=IP1234-NPB12345_00 result=RESULT_SUCCES console_id=0000000138e91b4e58236bf32besdafasdfasdfasdfsadf  account_id=11111  platform=pik
+```
+
+The below images depict the expression match on the data. Braces highlighted in green in the image below are the expression matches. There will be 7 matches in the current example.
```

Comment:
Technically they are not braces but rather parentheses; braces usually means "curly" brackets. The important point here is that each capturing group corresponds to one field value, so let's state it that way:

Braces ... matches. => The parentheses corresponding to capturing groups [capturing groups](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html#cg) are highlighted in green and each such group corresponds to one field of the POJO.